sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2919,6 +2919,17 @@ object SQLConf {
       .stringConf
       .createWithDefault("csv,json,orc,parquet")
 
+  val ADD_DEFAULT_COLUMN_EXISTING_TABLE_BANNED_PROVIDERS =
+    buildConf("spark.sql.defaultColumn.addColumnExistingTableBannedProviders")
+      .internal()
+      .doc("List of table providers wherein SQL commands are NOT permitted to assign DEFAULT " +
+        "values to new columns in existing tables, such as when using the ALTER TABLE ... " +
+        "ADD COLUMNS command in SQL. Comma-separated list, whitespace ignored, case-insensitive.")
```

Comment:
So there is no ENUM choice for this conf? Strings to specify table providers might not be robust (e.g. typos, etc.)

Reply:
I checked, it looks like not currently: today we support integers, floating-point numbers, booleans, strings, time values, and byte buffers [1].

I hear your concern and definitely open to ideas though :)

[1] https://src.dev.databricks.com/databricks/runtime@master/-/blob/core/src/main/scala/org/apache/spark/internal/config/ConfigBuilder.scala?L183&subtree=true

Reply:
We do support enum configurations via checking string values. You can check STORE_ASSIGNMENT_POLICY as an example.
For data sources, we can't make it enum since there are external data sources, e.g. Delta, BigQuery, etc.

Reply:
These are good points. It is not ideal to duplicate the table provider strings in two SQLConf entries.

@gengliangwang and @amaliujia what do you think about simply reusing `DEFAULT_COLUMN_ALLOWED_PROVIDERS` SQLConf and adding some simple rule to mark a table provider as "does not support `ALTER TABLE ADD COLUMN` commands"? For example, we could allow an asterisk after the table provider name to indicate this. For example, we could have `"csv,json,orc,parquet,myv2datasource*"` as the SQLConf value for a new V2 data source.

Reply:
To me, it's ok to have a mixed known list of data formats that might cover many usages (or re-use `DEFAULT_COLUMN_ALLOWED_PROVIDERS`) while leave room for unknown or new but working data format checking. 

I honestly don't have a good idea on this. Will also be happy to hear for different opinions from Spark community side.

Reply:
I ended up reusing `DEFAULT_COLUMN_ALLOWED_PROVIDERS` with an optional asterisk at the end of any table provider name to indicate that `ALTER TABLE ADD COLUMN` commands are not allowed. This seems to help reduce duplication by retaining only one SQLConf, I think.

Reply:
thanks! +1 to reuse the existing conf if there is no strong objection or obvious drawback.